keps/sig-node/2411-cri-container-log-rotation/README.md

```diff
@@ -0,0 +1,608 @@
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+- The kubelet assigns the log path for a container and runtime writes the container output to that path
+- The kubelet periodically checks the disk space occupied by the container logs and rotates them if necessary
+- After the logs are rotated, the kubelet sends a signal to the container runtime to re-open the log file
+- The kubelet exposes a consistent log directory structure with metadata so that any logging agent can integrate with it
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+- Shipping the logs directly to a remote storage
+- Supporting container runtimes that run on a different virtual/physical machine from the kubelet
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+Graduate the CRIContainerLogRotation feature gate from Beta to Stable. The kubelet already decides the container log directory structure and passes that down to the container runtime. The container runtime then writes the container's logs to this location. This makes the kubelet the best candidate to manage the rotation of the container logs as it already know the log directory structure and has access to it. The CRILogRotation feature gate implementation adds a container log manager package, which manages and rotates the container logs. It also adds 2 flags to the kubelet that allows the user to configure the maximum size of each log file and the maximum number of log files to have. These flags are **--container-log-max-size** and **--container-log-max-files**.
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+Users want to use a CRI container runtime and need their container logs to be managed and rotated by the kubelet.
```

Comment:
As a kubernetes user, I want container logs in to be managed and rotated ..

Reply:
Fixed